### DISCUSSION TITLE
Upgrade to borg 1.4 for Ubuntu 18.04

### DISCUSSION THREAD

**Discussion by televisi** [borgbackup/borg#8340] [General]
Hi,
Currently I'm using [docker-borgmatic](https://github.com/borgmatic-collective/docker-borgmatic) to automatically backup my files to Hetzner.
The docker-borgmatic is currently using borg version 1.4

The thing is, I'd like to run [borg check](https://borgbackup.readthedocs.io/en/stable/usage/check.html) using a different machine (Ubuntu 18.04.6 LTS).
When I checked "Current status of package in the repositories" page, Ubuntu 18.04 latest package is 1.1.15

Is it possible for me to install borg version 1.4 on my Ubuntu 18.04 machine?

As when I tried to install it, it installed version 1.1.15

> root@ubuntu: apt install borgbackup
> Reading package lists... Done
> Building dependency tree       
> Reading state information... Done
> borgbackup is already the newest version (1.1.15-1~ubuntu1.18.04.2).
> 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

Thank you!

**Comment by infectormp** [comment]
Try fatbinary from release section https://github.com/borgbackup/borg/releases

**Reply by televisi** [reply]
Thanks for your reply!
Sorry I'm a bit thick about this, I suspect I just need to download one of the version (e.g. borg-linux-glibc236) and follow the instructions on https://borgbackup.readthedocs.io/en/1.4-maint/installation.html#standalone-binary?

```
To install such a binary, just drop it into a directory in your PATH, make borg readable and executable for its users and then you can run borg:

sudo cp borg-linux-glibc236 /usr/local/bin/borg
sudo chown root:root /usr/local/bin/borg
sudo chmod 755 /usr/local/bin/borg

Optionally you can create a symlink to have borgfs available, which is an alias for borg mount:
ln -s /usr/local/bin/borg /usr/local/bin/borgfs
```

UPDATE: all good, found more explanations on "00_README.txt" file...

**Comment by televisi** [comment]
Okay, so I managed to follow the instructions to download "[borg-linux-glibc228](https://github.com/borgbackup/borg/releases/download/1.4.0/borg-linux-glibc228)" and ran GPG verification successfully.
> Good signature from "Thomas Waldmann [Email]

Note: I chose borg-linux-glibc228 because as per [this page](https://askubuntu.com/questions/445487/what-debian-version-are-the-different-ubuntu-versions-based-on) - Ubuntu 18.04 is classified as Debian 10.

But upon running it, it threw the following error:
> [3065] Error loading Python lib '/tmp/_MEIAT1rYC/libpython3.11.so.1.0': dlopen: /lib/x86_64-linux-gnu/libc.so.6: version 'GLIBC_2.28' not found (required by /tmp/_MEIAT1rYC/libpython3.11.so.1.0)

I also tried the following:
> apt-get install python3 python3-dev python3-pip python3-virtualenv libacl1-dev libacl1 libssl-dev liblz4-dev libzstd-dev libxxhash-dev build-essential pkg-config python3-pkgconfig

And looks like all of them are installed successfully too:

> Reading package lists... Done
> Building dependency tree       
> Reading state information... Done
> build-essential is already the newest version (12.4ubuntu1).
> libacl1 is already the newest version (2.2.52-3build1).
> libacl1-dev is already the newest version (2.2.52-3build1).
> python3-virtualenv is already the newest version (15.1.0+ds-1.1).
> liblz4-dev is already the newest version (0.0~r131-2ubuntu3.1).
> libssl-dev is already the newest version (1.1.1-1ubuntu2.1~18.04.23).
> libzstd-dev is already the newest version (1.3.3+dfsg-2ubuntu1.2).
> python3 is already the newest version (3.6.7-1~18.04).
> python3-dev is already the newest version (3.6.7-1~18.04).
> libxxhash-dev is already the newest version (0.7.3-1~ubuntu18.04.2).
> python3-pip is already the newest version (9.0.1-2.3~ubuntu1.18.04.8).
> 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

I suspect I need to install something else?

Thanks

**Reply by ThomasWaldmann** [reply]
Ubuntu 18.04 has glibc 2.27, that's why it does not work.

Also, it is out of support since 2023/04, so you should upgrade that anyway.

Maybe there are some borg 1.2.x fat binaries for very old glibcs, but not 1.4.x.

The steps with apt-get are NOT needed for the fat binary (but only for source-based installed including pip method).

**Comment by televisi** [comment]
Thanks for the reply!
Upgraded to ubuntu 24.10 and was able to run borg 1.4 using standalone binary file

**Reply by ThomasWaldmann** [reply]
For that ubuntu version, there is even a package of 1.4 in the maintainer ppa:

https://launchpad.net/~costamagnagianfranco/+archive/ubuntu/borgbackup?field.series_filter=noble
